From 4ff806d57f3c55993145f1dec5877d33634a14a7 Mon Sep 17 00:00:00 2001
From: Yingren Wang <yingrenw@codeaurora.org>
Date: Tue, 25 Dec 2018 13:47:42 +0800
Subject: [PATCH 35/45] SystemUI: Fix an issue where jumping to black after
 screen turned on

When press the power button several times, the screen will probabilistically jump from
the keyguard to the black screen after screen turned on. at this scene, previoud scrim
AOD state change triggered screen turned off is not performed until screen turned on.
Front ScrimView will display black color and covor keyguard in AOD state. so we can see
keyguard and then jump to black screen after screen turned on.

Ignore scrim AOD state change after screen turned on.

CRs-Fixed: 2342473
Change-Id: I7b206f0599420d2790b49f409a64fc264a4ecca0
---
 .../src/com/android/systemui/statusbar/phone/StatusBar.java    | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
index e30e2bae1c4..b1691c5d4c3 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/StatusBar.java
@@ -5110,7 +5110,7 @@ public class StatusBar extends SystemUI implements DemoMode, TunerService.Tunabl
 
         @Override
         public void startDozing() {
-            if (!mDozingRequested) {
+            if (!mDozingRequested && !isScreenTurningOnOrOn()) {
                 mDozingRequested = true;
                 DozeLog.traceDozing(mContext, mDozing);
                 updateDozing();
@@ -5159,6 +5159,7 @@ public class StatusBar extends SystemUI implements DemoMode, TunerService.Tunabl
                 DozeLog.traceDozing(mContext, mDozing);
                 mWakefulnessLifecycle.dispatchStartedWakingUp();
                 updateDozing();
+                updateScrimController();
             }
         }
 
-- 
2.17.1

